### PR TITLE
API: invite codes, auto-faucet

### PR DESCRIPTION
**Pull request by nalinbhardwaj**
Repurposes Faucet to auto-send for invite code accounts.

Onboarding: Moves invite page after flow selection page. So now if we have to test Use Existing flow on testnet we need to first go into create account and enter a testnet invite code, we can make this better in future but this only affects us internally for testing so doing this for now.

new flow screenshots:

Slight copy updates, "faucet" -> "team daimo", "Use existing" -> "Already have an account?"

### CODE REVIEW

File: apps/daimo-mobile/src/action/useCreateAccount.ts [hunk 1]
```diff
@@ -24,7 +25,7 @@ export function useCreateAccount(
   const exec = async () => {
     if (!pubKeyHex) return;
     setAS("loading", "Creating account...");
-    result.mutate({ name, pubKeyHex });
+    result.mutate({ name, pubKeyHex, invCode });
```
Comment (dcposch) [inline apps/daimo-mobile/src/action/useCreateAccount.ts:28]: 👍 👍

File: apps/daimo-mobile/src/view/screen/onboarding/OnboardingScreen.tsx [hunk 2]
```diff
@@ -281,13 +293,18 @@ function PageBubble({ count, index }: { count: number; index: number }) {
...
 }: {
   onNext: ({ isTestnet }: { isTestnet: boolean }) => void;
+  onPrev?: () => void;
   daimoChain: DaimoChain;
+  inviteCode: string;
+  setInviteCode: (inviteCode: string) => void;
 }) {
-  const [inviteCode, setInviteCode] = useState("");
-  const onChange = (text: string) => setInviteCode(text);
+  const onChange = (text: string) => setInviteCode(text.toLowerCase());
```
Comment (dcposch) [inline apps/daimo-mobile/src/view/screen/onboarding/OnboardingScreen.tsx:307]: 👍 nice then we don't have to list multiple invite codes in config

File: apps/daimo-mobile/src/view/screen/onboarding/OnboardingScreen.tsx [hunk 3]
```diff
@@ -383,7 +403,7 @@ function FlowSelectionPage({
       <Spacer h={16} />
       <ButtonBig
         type="subtle"
-        title="Use existing"
+        title="Already have an account?"
```
Comment (dcposch) [inline apps/daimo-mobile/src/view/screen/onboarding/OnboardingScreen.tsx:406]: much better 👍

File: packages/daimo-api/src/contract/faucet.ts [hunk 4]
```diff
@@ -31,31 +53,65 @@ export class Faucet {
...
+    if (!this.inviteCodes.has(invCode)) return false;
+    const { useCount, maxUses } = this.inviteCodes.get(invCode)!;
+    if (useCount >= maxUses) return false;
+    return true;
+  }
+
+  getStatus(address: Address, invCode: string): FaucetStatus {
+    if (!this.verifyInviteCode(invCode)) return "alreadySentInvite";
+
+    if (this.sent.has(address)) return "alreadySentAddress";
```
Comment (dcposch) [inline packages/daimo-api/src/contract/faucet.ts:70]: nit: `alreadySentCoins`/`alreadyRequestedCoins`

File: packages/daimo-api/src/env.ts [hunk 5]
```diff
@@ -264,7 +264,3 @@ export type ReadOnlyContractType<TAbi extends Abi> = GetContractReturnType<
   TAbi,
   PublicClient<Transport, Chain>
 >;
-
-export const daimoInviteCodes = new Set(
-  (process.env.DAIMO_INVITE_CODES || "zuconnect,devconnect").split(",")
-);
```
Comment (dcposch) [inline packages/daimo-api/src/env.ts:270]: 👍 👍

**Review by dcposch — APPROVED** [review]
LGreatTM

**Comment by nalinbhardwaj** [comment]
Added Clippy messages for low USDC balance.

File: packages/daimo-api/src/cron.ts [hunk 6]
```diff
@@ -86,17 +98,27 @@ export class Crontab {
...
+      functionName: "balanceOf",
+      args: [faucetAddr],
+    });
+    const balanceDollars = Number(amountToDollars(balanceUSDC));
+
+    await this.sendLowBalanceMessage(
+      balanceDollars,
+      `Faucet ${faucetAddr} USDC`,
+      200,
+      20
```
Comment (dcposch) [inline packages/daimo-api/src/cron.ts:120]: 👍

**Pull request merged**